woodchipper/context.py

```diff
@@ -94,19 +175,47 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self._token = None
         return False
 
+    def __call__(self, f):
+        self.decorator_mapping = _build_path_head_to_param_config_map(
+            self.injected_context, delimiter=self.path_delimiter
+        )
 
-logging_ctx = LoggingContextVar("logging_ctx")
+        @wraps(f)
+        def wrapper(*func_args, **func_kwargs):
 
+            # This will pair up the function signature parameters with the arguments that were passed. It's ...amazing.
+            # If we wanted to apply defaults as well (that weren't passed but were defined as defaults in the
+            # signature, there's a fairly easy process for that too through this api as well).
 
-def _convert_to_loggable_value(value: Any) -> LoggableValue:
-    loggable_types = {str, int, bool, Decimal, float, None}
-    if type(value) in loggable_types:
-        return cast(LoggableValue, value)
-    else:
-        try:
-            return str(value)
-        except Exception:
-            raise ValueError(
-                f"Value type (type={type(value)} is not a not a LoggableType ({str, int, bool, Decimal, float, None}) "
-                f"and cannot be converted to string."
-            )
+            # Note: We not will dig into the *args or **kwargs of the function, if the function definition used
+            # *args or **kwargs. It's more difficult for **kwargs and not possible with *args.
+            # TODO: attempt digging into function **kwargs as well?
+            # TODO: consider a configurable 'log_defaults' decorator parameter that will allow the logging of
```

Comment:
Should these be added as enhancements in the issues for the project?

Reply:
@Quidge What do you think?

Reply:
Yeah I definitely think a flag to log the defaults would be valuable.

I think there is less demand for the capability to dig into function kwargs. If an argument is in the `**kwargs` collection, it's likely that the function being decorated isn't particularly concerned with whatever is in there (else it'd be explicitly named and not in `kwargs`). There are cases where this isn't true, but they're less common.

I think it'd be worth an enhancement flag but it's not a high priority imo.